frontend/src/MuiForms/CustomTextInput.tsx

```diff
@@ -12,40 +14,48 @@ interface EntitySelectorProps {
   InputProps?: any
 }
 
-export default function EntitySelector(props: EntitySelectorProps) {
+export default function CustomTextInput(props: CustomTextInputProps) {
   const { onChange, value: currentValue, label, formContext } = props
 
-  const _onChange = (e) => {
-    onChange(e.target.value)
-  }
-
   const theme = useTheme()
 
+  const _onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value)
+  }
+
   return (
-    <TextField
-      {...props}
-      size='small'
-      sx={{
-        input: {
-          color: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-        label: {
-          WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-        '& .MuiInputBase-input.Mui-disabled': {
-          WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
-        },
-      }}
-      onChange={_onChange}
-      variant={!formContext.editMode ? 'standard' : 'outlined'}
-      label={label}
-      required={!formContext.editMode ? false : true}
-      value={currentValue || ''}
-      disabled={!formContext.editMode}
-      InputProps={{
-        ...props.InputProps,
-        disableUnderline: !formContext.editMode ? true : false,
-      }}
-    />
+    <Fragment key={label}>
+      <Typography sx={{ fontWeight: 'bold' }}>{label}</Typography>
+      <TextField
+        size='small'
+        sx={{
+          input: {
+            color: theme.palette.mode === 'light' ? 'black' : 'white',
+          },
+          label: {
+            WebkitTextFillColor: theme.palette.mode === 'light' ? 'black' : 'white',
+          },
+          '& .MuiInputBase-input.Mui-disabled': {
+            WebkitTextFillColor: currentValue
```

Comment:
This is very difficult to read. Nested ternaries are something we should avoid. Consider pulling out into a `useMemo`
```
const disabledWebkitTextFillColor = useMemo(() => {
  if (currentValue) {
    return theme.palette.mode === 'light' ? 'black' : 'white'
  } else {
    return theme.palette.mode === 'light' ? '#535353' : '#c8c8c8'
  }
}, [theme])
```